tests/chain_tests/delay_tests.cpp

```diff
@@ -1024,6 +1034,202 @@ BOOST_AUTO_TEST_CASE( link_delay_link_change_test ) { try {
+       ("quantity", "100.0000 CUR")
+       ("memo", "hi" )
+   );
+   BOOST_REQUIRE_EQUAL(transaction_receipt::executed, trace.status);
+   BOOST_REQUIRE_EQUAL(0, trace.deferred_transaction_requests.size());
+
+   chain.produce_blocks();
+
+   auto liquid_balance = get_currency_balance(chain, N(currency));
+   BOOST_REQUIRE_EQUAL(asset::from_string("999900.0000 CUR"), liquid_balance);
+   liquid_balance = get_currency_balance(chain, N(tester));
+   BOOST_REQUIRE_EQUAL(asset::from_string("100.0000 CUR"), liquid_balance);
+
+   // this transaction will be delayed 20 blocks
+   trace = chain.push_action(N(currency), name("transfer"), N(tester), fc::mutable_variant_object()
+       ("from", "tester")
+       ("to", "tester2")
+       ("quantity", "1.0000 CUR")
+       ("memo", "hi" ),
+       30, 10
+   );
+
+   BOOST_REQUIRE_EQUAL(transaction_receipt::delayed, trace.status);
+   BOOST_REQUIRE_EQUAL(1, trace.deferred_transaction_requests.size());
+   BOOST_REQUIRE_EQUAL(0, trace.action_traces.size());
+
+   chain.produce_blocks();
+
+   liquid_balance = get_currency_balance(chain, N(currency));
+   BOOST_REQUIRE_EQUAL(asset::from_string("999900.0000 CUR"), liquid_balance);
+   liquid_balance = get_currency_balance(chain, N(tester));
+   BOOST_REQUIRE_EQUAL(asset::from_string("100.0000 CUR"), liquid_balance);
+   liquid_balance = get_currency_balance(chain, N(tester2));
+   BOOST_REQUIRE_EQUAL(asset::from_string("0.0000 CUR"), liquid_balance);
+
+   try {
+      chain.push_action(config::system_account_name, contracts::unlinkauth::get_name(), tester_account, fc::mutable_variant_object()
+      ("account", "tester")
+      ("code", "currency")
+      ("type", "transfer"),
+      30, 7);
+      BOOST_FAIL("unlink should have a min delay set by previous linkauth");
+   } catch (...) { }
+
+   // this transaction will be delayed 20 blocks
+   chain.push_action(config::system_account_name, contracts::unlinkauth::get_name(), tester_account, fc::mutable_variant_object()
+           ("account", "tester")
+           ("code", "currency")
+           ("type", "transfer"),
+           30, 10
+   );
+   BOOST_REQUIRE_EQUAL(transaction_receipt::delayed, trace.status);
+   BOOST_REQUIRE_EQUAL(1, trace.deferred_transaction_requests.size());
+   BOOST_REQUIRE_EQUAL(0, trace.action_traces.size());
+
+   chain.produce_blocks();
+
+   liquid_balance = get_currency_balance(chain, N(tester));
+   BOOST_REQUIRE_EQUAL(asset::from_string("100.0000 CUR"), liquid_balance);
+   liquid_balance = get_currency_balance(chain, N(tester2));
+   BOOST_REQUIRE_EQUAL(asset::from_string("0.0000 CUR"), liquid_balance);
+
+   chain.produce_blocks(16);
+
+   liquid_balance = get_currency_balance(chain, N(tester));
+   BOOST_REQUIRE_EQUAL(asset::from_string("100.0000 CUR"), liquid_balance);
+   liquid_balance = get_currency_balance(chain, N(tester2));
+   BOOST_REQUIRE_EQUAL(asset::from_string("0.0000 CUR"), liquid_balance);
+
+   // this transaction will be delayed 20 blocks
```

Comment:
What does the second delayed transfer do for this test?  If it doesn't add to test coverage and just makes the test run longer, we should remove it.

Reply:
(I am basing my statement off of the exception you are expecting for the unlink above)